Library/Homebrew/linkage_checker.rb

```diff
@@ -83,6 +91,7 @@ def check_undeclared_deps
       next true if Formula[name].bin.directory?
       @brewed_dylibs.keys.map { |x| x.split("/").last }.include?(name)
     end
+    unnecessary_deps -= @broken_dylibs.map { |_, v| v.map { |d| dylib_to_dep(d) } }.flatten
```

Comment:
Split this into a few lines:
- Use a `do .. end` form each `map`
- Set a new variable which you do the `-=` with

Reply:
Also: you may be able to do `|, v|`

Reply:
At least locally, I need that `_` in this implementation: 
```
Error: /usr/local/Homebrew/Library/Homebrew/linkage_checker.rb:96: syntax error, unexpected ',', expecting '|'
    missing << @broken_dylibs.map do |, v|
```
I can use `.values` instead:
```ruby
    missing_deps = @broken_dylibs.values.map do |v|
      v.map do |d|
        dylib_to_dep(d)
      end
    end.flatten.compact
    unnecessary_deps -= missing_deps
```